src/lib.rs

```diff
@@ -65,6 +78,18 @@ impl From<&NixFile> for PathBuf {
     }
 }
 
+impl ToString for NixFile {
+    fn to_string(&self) -> String {
+        PathBuf::from(self).as_path().to_string_lossy().to_string()
+    }
+}
+
+impl From<String> for NixFile {
+    fn from(string: String) -> Self {
+        NixFile::Shell(PathBuf::from(string))
+    }
+}
```

Comment:
Where do we use this instance? It’s valid in theory, but going from a string to a PathBuf sounds like a design smell.

Reply:
This is more generally used in `internal_proto` - e.g. L246, where it's used to push the path into a `Reason::ProjectAdded`